src/Turing.jl

```diff
@@ -25,6 +25,25 @@ function setprogress!(progress::Bool)
     return progress
 end
 
+# Log density function
+struct LogDensityFunction{V,M,S,C}
```

Comment:
This looks great - I think we might want to move it and the callable object function below to DynamicPPL in the future.

Reply:
That were my thoughts as well. However, I think it would be good to (re-)use it in the optimization module and for DynamicHMC. I assumed it would be easier to do that before moving it to DynamicPPL since possibly it requires some minor changes. I thought it would be cleaner to do that in a follow-up PR and hence did not work on it in this PR.